### PR TITLE
fix: concurrent encryption exception for Android

### PR DESCRIPTION
**Pull request by ttypic**
Resolves https://github.com/ably/ably-flutter/issues/483
Jira link: [SDK-3871](https://ably.atlassian.net/browse/SDK-3871)

Bump the `ably-android` dependency to 1.2.32

[SDK-3871]: https://ably.atlassian.net/browse/SDK-3871?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -33,7 +33,7 @@ dependencies {
 }
 
 android {
-    compileSdkVersion 32
+    compileSdkVersion 33
```
Comment (sacOO7) [inline android/build.gradle:36]: Why do we need to increase this version?

Comment (ttypic) [reply android/build.gradle]: Sorry, it was shot in the dark, rollbacked it, I tried to fix Android tests, they are passing locally, but not in github-actions, and unfortunately I can find any way how to turn on human readable logs, they are generating 100K lines of logs it is absolutely impossible to understand what is wrong

File: .github/workflows/flutter_integration.yaml [hunk 2]
```diff
@@ -42,10 +42,10 @@ jobs:
         api-level: [21, 29]
       fail-fast: false
 
-    runs-on: 'ubuntu-latest'
```
Comment (ttypic) [inline .github/workflows/flutter_integration.yaml:45]: on Ubuntu Android 29 can't even start

see: 
- https://github.com/ReactiveCircus/android-emulator-runner/issues/46 
- https://github.com/ReactiveCircus/android-emulator-runner/issues/63#issuecomment-631848045

File: .github/workflows/flutter_integration.yaml [hunk 3]
```diff
@@ -39,17 +39,17 @@ jobs:
   android:
     strategy:
       matrix:
-        api-level: [21, 29]
+        api-level: [24, 29]
```
Comment (ttypic) [inline .github/workflows/flutter_integration.yaml:42]: 21 always fails with `java.lang.NoSuchMethodError: No virtual method setEndpointIdentificationAlgorithm` for a long time, it looks like we updated some library earlier that no longer support android 21

**Review by sacOO7 — APPROVED** [review]
LGTM

**Pull request merged**